core/src/View/Kokkos_BasicView.hpp

```diff
@@ -385,23 +385,67 @@ class BasicView {
   }
 
  public:
+  // check for accessor_arg for managed view ctors
+  template <class... P>
+  explicit inline BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer &&
+                           !Impl::ViewCtorProp<P...>::has_accessor_arg,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
+      : BasicView(arg_prop, arg_mapping, accessor_type{}) {}
+
+  template <class... P>
+
+  explicit inline BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer &&
+                           Impl::ViewCtorProp<P...>::has_accessor_arg,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
+      : BasicView(
+            arg_prop, arg_mapping,
+            accessor_type{
+                Impl::get_property<Impl::AccessorArgTag>(arg_prop).value()}) {}
+
+  // check for accessor_arg for unmanaged view ctors
+  template <class... P>
+  explicit KOKKOS_FUNCTION BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<Impl::ViewCtorProp<P...>::has_pointer &&
+                           !Impl::ViewCtorProp<P...>::has_accessor_arg,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
+      : BasicView(arg_prop, arg_mapping, accessor_type{}) {}
+
+  template <class... P>
+  explicit KOKKOS_FUNCTION BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<Impl::ViewCtorProp<P...>::has_pointer &&
+                           Impl::ViewCtorProp<P...>::has_accessor_arg,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
+      : BasicView(
+            arg_prop, arg_mapping,
+            accessor_type{
+                Impl::get_property<Impl::AccessorArgTag>(arg_prop).value()}) {}
+
   template <class... P>
   // requires(!Impl::ViewCtorProp<P...>::has_pointer)
   explicit inline BasicView(
       const Impl::ViewCtorProp<P...> &arg_prop,
       std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer,
-                       typename mdspan_type::mapping_type> const &arg_mapping)
-      : BasicView(create_data_handle(arg_prop, arg_mapping), arg_mapping) {}
+                       typename mdspan_type::mapping_type> const &arg_mapping,
+      const accessor_type &arg_accessor)
+      : BasicView(create_data_handle(arg_prop, arg_mapping), arg_mapping,
+                  arg_accessor) {}
```

Comment:
Do we need that constructor that takes directly the accessor as trailing argument or can we live with the ViewCtorProp that holds an accessor?

What if arg_prop has an accessor arg?  Why is it ok to just ignore it?
(Same question below)

Reply:
This constructor should have a constraint that arg_prop doesn't have an accessor arg

Reply:
ViewCtorProp doesn't currently hold accessors - and it also doesn't make sense. I mean in my mind it's the same as saying we pass mapping through CtorProp.  Also BasicView at its most fundamental is created from data_handle, mapping, and accessor - its constructors should express that.

Reply:
Or we add another 4 or 8 ctors to deal with the combinatorical situation

Reply:
By duplicating the data_handle creation logic, and removing again the ability to call constructors such as `View<..>(view_alloc("A"), mapping, accessor)` (which we do not have in develop) I was able to disallow the simultaneous existence of CtorProp with AccessorArg and an accessor, other than in a private ctor.

Reply:
Maybe I don't understand but I still don't understand why this ctor does *not* need `!Impl::ViewCtorProp<P...>::has_accessor_arg`

Reply:
its a private one which only is getting called from the the one which had an accessor_arg